### ISSUE TITLE
Expose line number in Context.getPosInfos()

### ISSUE THREAD

**Issue by kevinresol** [HaxeFoundation/haxe#5827]
It would be useful for converting a compile time `Position` to a runtime `haxe.PosInfos`

**Comment by benmerckx** [comment]
And the character range on that line would also be very useful. What I do now is regex the string representation of a `Position` to get that (`~/([0-9]+): characters ([0-9]+)-([0-9]+)/`), so I think the information should be available somewhere. But it wouldn't fit into `PosInfos` anymore.

**Comment by kevinresol** [comment]
> (~/([0-9]+): characters ([0-9]+)-([0-9]+)/)

Wow, that looks promising. I will give it a try.

**Comment by nadako** [comment]
note that the real regexp is a bit more complicated, since haxe can report line span as well, see e.g. vshaxe problem matcher https://github.com/vshaxe/vshaxe/blob/master/scaffold/.vscode/tasks.json#L8

**Comment by frabbit** [comment]
that would be nice, something like `PositionTools.getLine` would be enough.

**Comment by kevinresol** [comment]
If there is some API like `Context.toPosExpr(pos:Position):ExprOf<PosInfos>` then would be awesome.

**Comment by RealyUniqueName** [comment]
I believe this issue is resolved with #6599
```haxe
var line = haxe.macro.PositionTools.toRange(haxe.macro.Context.currentPos()).start.line;
```

**Comment by PXshadow** [comment]
If anyone else stumbles on this Haxe 4.2 version you write it now as: 
```haxe
var line = haxe.macro.PositionTools.toLocation(haxe.macro.Context.currentPos()).range.start.line;
```
